app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java

```diff
@@ -18,52 +35,309 @@
+        EditText timeText = putHourTimeLayout(time);
+        timeText.setContentDescription(getString(R.string.timeText));
+        layout.addView(time);
+        Spinner sp = initializeSpinner(layout);
+        sp.setContentDescription(getString(R.string.spinnerText));
+        dialogElements(newPersonal, layout, reasonText, dateText, timeText, sp);
+    }
+
+    /**
+     * Initialize the dialog elements.
+     * @param timeText The time
+     * @param sp The spinner
+     * @param reasonText The event description
+     * @param dateText The event date
+     * @param layout The layout of the dialog
+     * @param newPersonal The dialog
+     */
+    private void dialogElements(MaterialAlertDialogBuilder newPersonal, LinearLayout layout, EditText reasonText,
+                                TextView dateText, EditText timeText, Spinner sp) {
+        newPersonal.setTitle(getString(R.string.dialog_new_event));
+        newPersonal.setMessage(R.string.dialog_new_event_message);
+        newPersonal.setView(layout);
+        cancelDialog(newPersonal);
+        createPersonalEventListener(newPersonal, reasonText, dateText, timeText, sp);
+    }
+
+    /**
+     * Listener of the create button of the dialog.
+     * @param newPersonal The dialog
+     * @param dateText The date of the event
+     * @param reasonText The reason of the event
+     * @param sp The spinner for pet selection
+     * @param timeText The hour of the event
+     */
+    private void createPersonalEventListener(MaterialAlertDialogBuilder newPersonal, EditText reasonText, TextView
+            dateText, EditText timeText, Spinner sp) {
+        newPersonal.setPositiveButton(getString(R.string.create), (dialog, which) -> {
+            if (sp.getSelectedItem() != null) {
+                if (reasonText.getText().toString().length() != 0) {
+                    createPersonalEvent(reasonText, dateText, timeText, sp);
+                } else {
+                    toastText(getString(R.string.no_description));
+                }
+            } else {
+                toastText(getString(R.string.add_a_pet));
+            }
+        });
+    }
+
+    /**
+     * Create a new toast.
+     * @param text The toast text
+     */
+    private void toastText(String text) {
+        Toast toast1 = Toast.makeText(getActivity(), text, Toast.LENGTH_LONG);
+        toast1.setGravity(Gravity.CENTER, 0, 0);
+        toast1.show();
+    }
+
+    /**
+     * Create a new personal event.
+     * @param dateText The date of the event
+     * @param reasonText The reason of the event
+     * @param sp The spinner for pet selection
+     * @param timeText The hour of the event
+     */
+    private void createPersonalEvent(EditText reasonText, TextView dateText, EditText timeText, Spinner sp) {
+        String petName = sp.getSelectedItem().toString();
+        StringBuilder dateTime = new StringBuilder(dateText.getText());
+        dateTime.append('T');
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497)